### PR TITLE
Assumptions for work depth analysis

### PR DESCRIPTION
**Pull request by hodelcl**
This PR enables the user to enter assumptions for the symbols before calling the work depth analysis.

It uses a VS code `InputBox` to handle the input. Validity of the input is checked by a regex. If the check passes, the assumptions are forwarded to the work depth analysis.

The corresponding PR to the dace repo can be found [here](https://github.com/spcl/dace/pull/1363).

### CODE REVIEW

File: src/components/dace_interface.ts [hunk 1]
```diff
@@ -909,20 +939,36 @@ implements vscode.WebviewViewProvider {
...
-                    },
-                    (data: any) => {
-                        resolve(data.avgParallelismMap);
-                        DaCeInterface.getInstance()?.hideSpinner();
-                    },
-                    (error: any) => {
-                        this.genericErrorHandler(error.message, error.details);
-                        reject(error.message);
+
+                vscode.window.showInputBox({
```
Comment (phschaad) [inline src/components/dace_interface.ts:943]: This is quite some code repetition, is it possible to factor this out into a separate small function? (Showing the prompt, checking for the format, showing error message if violated, etc.)

File: backend/run_dace.py [hunk 2]
```diff
@@ -53,7 +53,7 @@
 
 sys.path.append(path.abspath(path.dirname(__file__)))
 
-from dace_vscode import arith_ops, depth, transformations
+from dace_vscode import depth, transformations
```
Comment (phschaad) [inline backend/run_dace.py:56]: Can't the `dace_vscode.arith_ops` module be removed entirely now?

File: backend/dace_vscode/depth.py [hunk 3]
```diff
@@ -2,23 +2,45 @@
...
 import sympy as sp
-from dace.sdfg.work_depth_analysis.work_depth import analyze_sdfg, get_tasklet_work_depth, get_tasklet_avg_par
+from dace.sdfg.work_depth_analysis.work_depth import analyze_sdfg, get_tasklet_work, get_tasklet_work_depth, get_tasklet_avg_par
 
 from dace_vscode.utils import (
     load_sdfg_from_json,
     get_exception_message,
 )
 
+def get_work(sdfg_json, assumptions):
```
Comment (phschaad) [inline backend/dace_vscode/depth.py:12]: This module should now probably be renamed to `work_depth` or something :-)

**Review by phschaad — CHANGES_REQUESTED** [review]
Very nice, thank you :-) Small changes only

**Comment by hodelcl** [comment]
I integrated all your comments:

- put the `InputBox` for the assumptions and the input checking in separate functions,
- deleted the `dace_vscode.arith_ops` module,
- renamed the `depth` module to `work_depth`.

**Review by phschaad — APPROVED** [review]
LGTM, thanks

**Pull request merged**